### DISCUSSION TITLE
Is it possible to have multiple addresses connected simultaneously?

### DISCUSSION THREAD

**Discussion by jcislo** [rainbow-me/rainbowkit#1961] [General]
I am building a nextjs front-end and the desired functionality is to display multiple wallet addresses at once and also have the ability to sign transactions from all addresses on-screen without needing to enter metamask (or whichever wallet the user connected from) to switch to that particular address. 

Is this at all possible using Rainbowkit? Perhaps a modification could be made to create multiple viem "walletClient" objects?

**Comment by 0xmkh** [comment]
Hey @jcislo. RainbowKit doesn't support connecting multiple wallets at once and getting multiple connected addresses. You could build your own flow where you can connect multiple wallets at once and get multiple connected addresses and use wagmi hook called [useConnection](https://wagmi.sh/react/api/hooks/useConnections#useconnections).

Let me know if this answers your question 👍
